### PR TITLE
Database migration from multiple locations

### PR DESCRIPTION
**Pull request by AnonymousWalker**
This change is [Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/655)

### CODE REVIEW

**Review by jsarabia — CHANGES_REQUESTED** [review]
> Reviewed 1 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/655)* status: 1 of 9 files reviewed, 1 unresolved discussion (waiting on @AnonymousWalker and @jsarabia)

___
*[`jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/database/AppDatabase.kt` line 35 at r3](https://reviewable.io/reviews/bible-translation-tools/orature/655#-N9fzToQ0f0TeBbStvRu:-N9fzToQ0f0TeBbStvRv:b-38zap) ([raw file](https://github.com/bible-translation-tools/orature/blob/3bfb06ac2851ace1cc2f868a68ae6592c5ffa8b0/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/database/AppDatabase.kt#L35)):*
> ```Kotlin
> const val CREATION_SCRIPT = "sql/CreateAppDb.sql"
> 
> class AppDatabase(
> ```

Can you please explain to me what this class is and its purpose?
Then can you explain why the added functionality belongs here, not somewhere else?

**Comment by AnonymousWalker** [comment]
*[`jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/database/AppDatabase.kt` line 35 at r3](https://reviewable.io/reviews/bible-translation-tools/orature/655#-N9fzToQ0f0TeBbStvRu:-N9g132OCchsdd-2OERx:b-y7895b) ([raw file](https://github.com/bible-translation-tools/orature/blob/3bfb06ac2851ace1cc2f868a68ae6592c5ffa8b0/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/database/AppDatabase.kt#L35)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

Can you please explain to me what this class is and its purpose?
Then can you explain why the added functionality belongs here, not somewhere else?
</blockquote></details>

yeah... AppDatabase is acting like a datasource or repository implementation, it shouldn't "know" migration stuff like this

**Review by jsarabia — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/655)* status: 1 of 11 files reviewed, 3 unresolved discussions (waiting on @AnonymousWalker and @jsarabia)

___
*[`jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/database/AppDatabase.kt` line 139 at r5](https://reviewable.io/reviews/bible-translation-tools/orature/655#-N9mUwXq6Lll3P2eVdo3:-N9mUwXq6Lll3P2eVdo4:bl3sp3m) ([raw file](https://github.com/bible-translation-tools/orature/blob/b0039e17a061fa33d6eb3e869fe658d32f796392/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/database/AppDatabase.kt#L139)):*
> ```Kotlin
>             val sqLiteDataSource = SQLiteDataSource()
>             sqLiteDataSource.url = "jdbc:sqlite:${databaseFile.toURI().path}"
>             sqLiteDataSource.config.toProperties().setProperty("foreign_keys", "true")
> ```

can this be a function which is used in the above code as well?
___
*[`jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/database/AppDatabase.kt` line 151 at r5](https://reviewable.io/reviews/bible-translation-tools/orature/655#-N9mUfa46qRkIrkk25AU:-N9mUfa46qRkIrkk25AV:b19qbq6) ([raw file](https://github.com/bible-translation-tools/orature/blob/b0039e17a061fa33d6eb3e869fe658d32f796392/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/database/AppDatabase.kt#L151)):*
> ```Kotlin
>                         }
>                 } catch (e: DataAccessException) {
>                     -1
> ```

if this is invalid, should this be null instead? a -1 could accidentally be migrated since migration checks if the version is lower than the schema, and -1 always will be.

Basically, we don't necessarily know when we might end up using a "getDatabaseVersion" method somewhere else in the future

**Review by AnonymousWalker — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/655)* status: 1 of 11 files reviewed, 3 unresolved discussions (waiting on @AnonymousWalker and @jsarabia)

___
*[`jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/database/AppDatabase.kt` line 139 at r5](https://reviewable.io/reviews/bible-translation-tools/orature/655#-N9mUwXq6Lll3P2eVdo3:-N9mYltC1VcPPWZA9mBZ:b-896fix) ([raw file](https://github.com/bible-translation-tools/orature/blob/b0039e17a061fa33d6eb3e869fe658d32f796392/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/database/AppDatabase.kt#L139)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

can this be a function which is used in the above code as well?
</blockquote></details>

Done.
___
*[`jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/database/AppDatabase.kt` line 151 at r5](https://reviewable.io/reviews/bible-translation-tools/orature/655#-N9mUfa46qRkIrkk25AU:-N9mYm44BO2AX131gXxq:b-896fix) ([raw file](https://github.com/bible-translation-tools/orature/blob/b0039e17a061fa33d6eb3e869fe658d32f796392/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/database/AppDatabase.kt#L151)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

if this is invalid, should this be null instead? a -1 could accidentally be migrated since migration checks if the version is lower than the schema, and -1 always will be.

Basically, we don't necessarily know when we might end up using a "getDatabaseVersion" method somewhere else in the future
</blockquote></details>

Done.

**Review by jsarabia — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/655#-:-N9mbWFt1EvSPS40kh34:bnfp4nl)* status: 1 of 11 files reviewed, 2 unresolved discussions (waiting on @AnonymousWalker and @jsarabia)
